src/Tmds.DBus/DBusConnection.cs

```diff
@@ -39,7 +39,7 @@ public void Dispose()
 
         private class NameOwnerWatcherRegistration : IDisposable
         {
-            public NameOwnerWatcherRegistration(DBusConnection dbusConnection, string key, OwnerChangedMatchRule rule, Action<ServiceOwnerChangedEventArgs, Exception> handler)
+            public NameOwnerWatcherRegistration(DBusConnection dbusConnection, string key, SignalMatchRule rule, Action<ServiceOwnerChangedEventArgs, Exception> handler)
```

Comment:
Do we still need `NameOwnerWatcherRegistration` and `_nameOwnerWatchers`?

Reply:
It looked like it has some special rules for handling `".*"` in the passed in service name, so I left it as-is.

Reply:
Ok, I'm fine keeping these. I may take a look at removing them in a follow-up PR.